nunaliit2-js/src/main/js/nunaliit2/n2.modelUtils.js

```diff
@@ -497,7 +497,579 @@ var ModelIntersect = $n2.Class({
+				doc = sourceState.removed[i];
+				docId = doc._id;
+
+				if (doc.nunaliit_schema) {
+					schema = doc.nunaliit_schema;
+
+					schemaDocs = this.schemaDocsByDocId[schema];
+					delete schemaDocs[docId];
+					delete this.docInfosByDocId[docId];
+					this.removedMap[docId] = doc;
+					processingRequired = true;
+				}
+			}
+		}
+
+		if (processingRequired) {
+			if (!this.modelIsLoading) {
+				this._joinSchemaDocs();
+			}
+
+			// Report changes
+			added = $n2.utils.values(this.addedMap);
+			updated = $n2.utils.values(this.updatedMap);
+			removed = $n2.utils.values(this.removedMap);
+			this._reportStateUpdate(added, updated, removed);
+		}
+
+	},
+
+	// Entry function to start copy to schema document transformations
+	_joinSchemaDocs: function() {
+		var docId, docInfo, doc, transform, i, copyToSchemaDocIds;
+		var added, updated, removed;
+		this.copyToSchemaDocsByDocId = this.schemaDocsByDocId[this.copyToSchema] || {};
+
+		copyToSchemaDocIds = Object.keys(this.copyToSchemaDocsByDocId);
+
+		// Loop over all documents, recomputing doc transforms
+		for (i = 0; i < copyToSchemaDocIds.length; i += 1) {
+			docId = copyToSchemaDocIds[i];
+			docInfo = this.copyToSchemaDocsByDocId[docId];
+			doc = docInfo.joined || docInfo.original;
+
+			if (Object.hasOwnProperty.call(this.schemaDocsByDocId, this.copyFromSchema)) {
+				this.copyFromSchemaDocsByDocId = this.schemaDocsByDocId[this.copyFromSchema] || {};
+
+				// Check that copyFrom dictionary has items
+				if (Object.keys(this.copyFromSchemaDocsByDocId).length) {
+					// transform the document
+					transform = this._computeTransform(doc);
+
+					if (doc === transform) {
+						// No transform
+					} else {
+						// A transform was computed
+						docInfo.doc = undefined;
+						docInfo.doc = transform;
+
+						// Update copy to schema original doc for future joins
+						docInfo.joined = transform;
+						this.copyToSchemaDocsByDocId[docId] = docInfo;
+
+						// Find where it belongs
+						if (this.addedMap[docId]) {
+							this.addedMap[docId] = transform;
+						} else if (this.updatedMap[docId]) {
+							this.updatedMap[docId] = transform;
+						} else if (this.removedMap[docId]) {
+							// just leave previous
+						} else {
+							// Not already in list. Add to update list
+							this.updatedMap[docId] = transform;
+						}
+					}
+				}
+			}
+		}
+
+		// Perform next schema docs join.
+		if (this.joinNum < this.joins.length - 1) {
+			this.joinNum += 1;
+			this._setSchemasToJoin(this.joins[this.joinNum]);
+			this._joinSchemaDocs();
+
+		} else {
+			// Report changes if no more transformations are needed.
+			added = $n2.utils.values(this.addedMap);
+			updated = $n2.utils.values(this.updatedMap);
+			removed = $n2.utils.values(this.removedMap);
+			this._reportStateUpdate(added, updated, removed);
+		}
+	},
+
+	// Checks if an object is a Nunaliit reference object
+	_isNunaliitRefObj: function(obj) {
```

Comment:
Another util function?  Only if it happens to be used elsewhere.

Reply:
Currently this is the only place this is used, but if we use it in the future, we should add it as a util function.